### PR TITLE
More blocking functions

### PR DESCRIPTION
**Pull request by navinrathore**
#261

### CODE REVIEW

**Comment by navinrathore** [comment]
Tested findTrainingData, Train etc. with 1 field for integer, double.
separate commits  for different type of blocking functions. They can be views commit wise.

File: core/src/main/java/zingg/hash/TruncateDouble.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package zingg.hash;
+
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.api.java.UDF1;
+import org.apache.spark.sql.types.DataTypes;
+
+public class TruncateDouble extends HashFunction implements UDF1<Double, Double> {
+	int numDecimalPlaces;
+	static final int[] POWERS_OF_10 = {1, 10, 100, 1000, 10000, 100000};
```
Comment (sonalgoyal) [inline core/src/main/java/zingg/hash/TruncateDouble.java:9]: why use this? just say 10 exp the argument.

File: core/src/test/java/zingg/hash/TestRangeDbl.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		value = new RangeBetween1000And10000Dbl();
+		assertEquals(0, value.call((double) 65));
+	}
+
+	@Test
+	public void testRangeForValue867() {
+		RangeDbl value = new RangeBetween0And10Dbl();
+		assertEquals(0, value.call((double) 867));
+		value = new RangeBetween10And100Dbl();
+		assertEquals(0, value.call((double) 867));
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/hash/TestRangeDbl.java:76]: saying 867d makes the code more readable..upto you if you feel that way.

File: core/src/test/java/zingg/hash/TestRangeDbl.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package zingg.hash;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/hash/TestRangeDbl.java:5]: tests for different classes should be in different junit classes. multiple asserts should not go into same unit test, as you dont know then which one broke. If you think it is too much to write junits for the derived classes, just test the main base class.

File: core/src/test/java/zingg/hash/TestRangeInt.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
+package zingg.hash;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestRangeInt {
+
+	@Test
+	public void testRangeForValueZero() {
```
Comment (sonalgoyal) [inline core/src/test/java/zingg/hash/TestRangeInt.java:10]: same comment as earlier

**Review by sonalgoyal — CHANGES_REQUESTED** [review]

File: core/src/main/java/zingg/hash/TruncateDouble.java (hunk 1 above)
Comment (navinrathore) [reply core/src/main/java/zingg/hash/TruncateDouble.java:9]: pow() seems a relatively expansive operation.
https://stackoverflow.com/questions/46983772/fastest-way-to-obtain-a-power-of-10

File: core/src/test/java/zingg/hash/TestRangeDbl.java (hunk 2 above)
Comment (navinrathore) [reply core/src/test/java/zingg/hash/TestRangeDbl.java:76]: Shall Make the changes

File: core/src/test/java/zingg/hash/TestRangeDbl.java (hunk 3 above)
Comment (navinrathore) [reply core/src/test/java/zingg/hash/TestRangeDbl.java:5]: I'll change

File: core/src/test/java/zingg/hash/TestRangeInt.java (hunk 4 above)
Comment (navinrathore) [reply core/src/test/java/zingg/hash/TestRangeInt.java:10]: I'll make change

**Pull request merged**